eng/pipelines/docindex.yml

```diff
@@ -34,16 +35,39 @@ jobs:
           versionSpec: '3.6'
       - template: /eng/common/pipelines/templates/steps/mashup-doc-index.yml
         parameters:
-          SourceDirectory: $(Build.SourcesDirectory)        
-      - task: CopyFiles@2
-        displayName: Copy HTML to Artifacts Directory
-        inputs:
-          sourceFolder: $(Build.SourcesDirectory)/docfx_project/
-          content: '**\*'
-          targetFolder: $(Build.ArtifactStagingDirectory)/docfx_project
-          overWrite: true
+          SourceDirectory: $(Build.ArtifactStagingDirectory)
+      - pwsh: |
+          Copy-Item -Path $(Build.SourcesDirectory)/eng/common/scripts/git-branch-push.ps1 -Destination ./ -Recurse -Force
+          echo "##vso[task.setvariable variable=gitPushScriptPath]$(Build.BinariesDirectory)/git-branch-push.ps1"
+        workingDirectory: $(Build.BinariesDirectory)
+        displayName: Move Git Push scripts to Tool Directory  
+
       - task: PublishPipelineArtifact@0
         condition: succeeded()
         inputs:
           artifactName: "Doc.Index"
-          targetPath: $(Build.ArtifactStagingDirectory)/docfx_project/_site
\ No newline at end of file
+          targetPath: $(Build.ArtifactStagingDirectory)/docfx_project/_site
+          
+      - pwsh: |
+          git checkout -b gh-pages-local --track origin/gh-pages-root
+        workingDirectory: $(Build.SourcesDirectory)
+        displayName: Git pull GH pages branch
+
+      - pwsh: |
+          Copy-Item -Path $(Build.ArtifactStagingDirectory)/docfx_project/_site/* -Destination ./ -Recurse -Force
+          git add -A
```

Comment:
We shouldn't need this explicit add.

Reply:
All files copied to the gh-pages-local branch are untracked from git. git-branch-push will create a new branch which inherit gh-pages-local which did not include the new changes.

Reply:
Yes but when we do the commit we pass in the "-am" option with the "-a" part means include all changes.  Is that not picking up new files? If it is not the doing the "git add -A"

Reply:
They are not the same branch. git-branch-push is on gh-pages which inherited from gh-page-local I am creating. However, the changes are untracked in gh-pages-local if no `git add -A`